[Bug]: `stampRecordTypeMultiDimensionMap` unit test fails for jacoco build

### Description

$subject



### Steps to Reproduce

- run `stampRecordTypeMultiDimensionMap` unit test
- the test passes. But the console log shows the following stack trace

```
java.lang.instrument.IllegalClassFormatException: Error while instrumenting map-stamp-expr-test with JaCoCo 0.8.9.202303310957/c0ad781.
	at org.jacoco.agent.rt.internal_4481564.CoverageTransformer.transform(CoverageTransformer.java:94)
	at java.instrument/java.lang.instrument.ClassFileTransformer.transform(ClassFileTransformer.java:244)
	at java.instrument/sun.instrument.TransformerManager.transform(TransformerManager.java:188)
	at java.instrument/sun.instrument.InstrumentationImpl.transform(InstrumentationImpl.java:541)
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150)
	at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:524)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:427)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:421)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:420)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525)
	at org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:180)
	at org.ballerinalang.test.BRunUtil.invokeOnJBallerina(BRunUtil.java:153)
	at org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:92)
	at org.ballerinalang.test.BRunUtil.invoke(BRunUtil.java:278)
	at org.ballerinalang.test.expressions.stamp.MapStampInbuiltFunctionTest.testStampAnydataMapToAnydata(MapStampInbuiltFunctionTest.java:231)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
	at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.io.IOException: Error while instrumenting map-stamp-expr-test with JaCoCo 0.8.9.202303310957/c0ad781.
	at org.jacoco.agent.rt.internal_4481564.core.instr.Instrumenter.instrumentError(Instrumenter.java:161)
	at org.jacoco.agent.rt.internal_4481564.core.instr.Instrumenter.instrument(Instrumenter.java:111)
	at org.jacoco.agent.rt.internal_4481564.CoverageTransformer.transform(CoverageTransformer.java:92)
	... 30 more
Caused by: org.jacoco.agent.rt.internal_4481564.asm.MethodTooLargeException: Method too large: map-stamp-expr-test.stampRecordTypeMultiDimensionMap (Lio/ballerina/runtime/internal/scheduling/Strand;)Ljava/lang/Object;
	at org.jacoco.agent.rt.internal_4481564.asm.MethodWriter.computeMethodInfoSize(MethodWriter.java:2088)
	at org.jacoco.agent.rt.internal_4481564.asm.ClassWriter.toByteArray(ClassWriter.java:511)
	at org.jacoco.agent.rt.internal_4481564.core.instr.Instrumenter.instrument(Instrumenter.java:92)
	at org.jacoco.agent.rt.internal_4481564.core.instr.Instrumenter.instrument(Instrumenter.java:109)
	... 31 more
```

### Affected Version(s)

current master

### OS, DB, other environment details and versions

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

https://github.com/ballerina-platform/ballerina-lang/issues/41052

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

--- a/tests/jballerina-unit-test/src/test/resources/test-src/expressions/stamp/map-stamp-expr-test.bal
+++ b/tests/jballerina-unit-test/src/test/resources/test-src/expressions/stamp/map-stamp-expr-test.bal
@@ -206,6 +206,13 @@ function stampRecordTypeMultiDimensionMap() {
     map<Teacher> teacherMap = {"a": p1, "b": p2};
     map<map<Teacher>> multiMap = {"aa": teacherMap, "bb": teacherMap};
     map<map<Employee>> mapValue = checkpanic multiMap.cloneWithType(EmployeeDoubleMap);
+    assertMapValue_aa(mapValue);
+    assertMapValue_bb(mapValue);
+    test:assertEquals(mapValue.length(), 2);
+    test:assertEquals((typeof mapValue).toString(), "typedesc EmployeeDoubleMap");
+}
+
+function assertMapValue_aa(map<map<Employee>> mapValue) {
     test:assertEquals(mapValue["aa"]["a"]["name"], "Raja");
     test:assertEquals(mapValue["aa"]["a"]["age"], 25);
     test:assertEquals(mapValue["aa"]["a"]["status"], "single");
@@ -216,6 +223,9 @@ function stampRecordTypeMultiDimensionMap() {
     test:assertEquals(mapValue["aa"]["b"]["status"], "single");
     test:assertEquals(mapValue["aa"]["b"]["batch"], "LK2014");
     test:assertEquals(mapValue["aa"]["b"]["school"], "Hindu College");
+}
+
+function assertMapValue_bb(map<map<Employee>> mapValue) {
     test:assertEquals(mapValue["bb"]["a"]["name"], "Raja");
     test:assertEquals(mapValue["bb"]["a"]["age"], 25);
     test:assertEquals(mapValue["bb"]["a"]["status"], "single");
@@ -226,8 +236,6 @@ function stampRecordTypeMultiDimensionMap() {
     test:assertEquals(mapValue["bb"]["b"]["status"], "single");
     test:assertEquals(mapValue["bb"]["b"]["batch"], "LK2014");
     test:assertEquals(mapValue["bb"]["b"]["school"], "Hindu College");
-    test:assertEquals(mapValue.length(), 2);
-    test:assertEquals((typeof mapValue).toString(), "typedesc EmployeeDoubleMap");
 }
 
 function stampAnydataToIntMultiDimensionMap() returns map<map<map<int>>>|error {

--- a/tests/jballerina-unit-test/src/test/resources/test-src/expressions/stamp/map-stamp-expr-test.bal
+++ b/tests/jballerina-unit-test/src/test/resources/test-src/expressions/stamp/map-stamp-expr-test.bal
@@ -206,13 +206,13 @@ function stampRecordTypeMultiDimensionMap() {
     map<Teacher> teacherMap = {"a": p1, "b": p2};
     map<map<Teacher>> multiMap = {"aa": teacherMap, "bb": teacherMap};
     map<map<Employee>> mapValue = checkpanic multiMap.cloneWithType(EmployeeDoubleMap);
-    assertMapValue_aa(mapValue);
-    assertMapValue_bb(mapValue);
+    assertMapValuePart1(mapValue);
+    assertMapValuePart2(mapValue);
     test:assertEquals(mapValue.length(), 2);
     test:assertEquals((typeof mapValue).toString(), "typedesc EmployeeDoubleMap");
 }
 
-function assertMapValue_aa(map<map<Employee>> mapValue) {
+function assertMapValuePart1(map<map<Employee>> mapValue) {
     test:assertEquals(mapValue["aa"]["a"]["name"], "Raja");
     test:assertEquals(mapValue["aa"]["a"]["age"], 25);
     test:assertEquals(mapValue["aa"]["a"]["status"], "single");
@@ -225,7 +225,7 @@ function assertMapValue_aa(map<map<Employee>> mapValue) {
     test:assertEquals(mapValue["aa"]["b"]["school"], "Hindu College");
 }
 
-function assertMapValue_bb(map<map<Employee>> mapValue) {
+function assertMapValuePart2(map<map<Employee>> mapValue) {
     test:assertEquals(mapValue["bb"]["a"]["name"], "Raja");
     test:assertEquals(mapValue["bb"]["a"]["age"], 25);
     test:assertEquals(mapValue["bb"]["a"]["status"], "single");

